mdagent/tools/base_tools/analysis_tools/hydrogen_bonding_tools.py

```diff
@@ -0,0 +1,97 @@
+import mdtraj as md
+from langchain.tools import BaseTool
+
+from mdagent.utils import PathRegistry, load_single_traj
+
+
+class BakerHubbard(BaseTool):
+    name = "Baker_hubbard"
+    description = """Identify hydrogen bonds based that are present in at least 10%
+     of each frames ( freq =0.1) and provides a list of tuples with each tuples
+    containing three  integers representing the indices of atoms (donor, hydrogen,
+    acceptor) involved in the hydrogen bonding."""
+
+    path_registry: PathRegistry | None = None
+
+    def __init__(self, path_registry: PathRegistry):
+        super().__init__()
+        self.path_registry = path_registry
+
+    def _run(self, traj_file, top_file=None, freq=0.1):
```

Comment:
Can you add some type hints for your input args? (ie. `traj_file:str`, `top_file:str|None=None`, etc.